traces

```diff
@@ -33,16 +34,120 @@ use Symfony\Component\Console\Style\SymfonyStyle;
+        $headerValue = $response->getHeader('Link')[0];
+        preg_match(REGEX, $headerValue, $matches);
+        $nbPages = $matches[1];
+    } else {
+        $nbPages = 1;
+    }
+
+    // Fetch pages
+    $io->progressStart($nbPages);
+    for ($i = 1; $i <= $nbPages; $i++) {
+        $response = $client->get($requestUri . '?page=' . $i, $authHeaders);
+        $repositories = json_decode($response->getBody(), true);
+        foreach ($repositories as $repository) {
+            if ($repository['archived'] || $repository['private']|| $repository['fork']) {
+                continue;
+            }
+            if (in_array($repository['full_name'], $excludeRepositories)) {
+                continue;
+            }
+            $orgRepositories[] = $repository['full_name'];
+        }
+        $io->progressAdvance();
+    }
+    $io->progressFinish();
+
+    return $orgRepositories;
+}
+
+function fetchRepositoryContributors(
+    Client $client,
+    array $authHeaders,
+    string $repository
+) {
+    $contributors = [];
+    $requestUri = str_replace('{repository}', $repository, GITHUB_CONTRIBUTORS_POINT);
+    $response = $client->get($requestUri, $authHeaders);
+    if ($response->getStatusCode() !== 200) {
+        return $contributors;
+    }
+    if ($response->hasHeader('Link')) {
+        $headerValue = $response->getHeader('Link')[0];
+        preg_match(REGEX, $headerValue, $matches);
+        $nbPages = $matches[1];
+    } else {
+        $nbPages = 1;
+    }
+
+    for ($i = 1; $i <= $nbPages; $i++) {
+        $response = $client->get($requestUri . '?page=' . $i, $authHeaders);
+        $contributors = array_merge(
+            $contributors,
+            json_decode($response->getBody(), true)
+        );
+    }
+    return $contributors;
+}
+
+/**
+ * @param string $file
+ * @return array
+ */
+function fetchConfiguration(string $file): array {
+    if (empty($file)) {
+        return [];
+    }
+    if (!file_exists($file) || !is_readable($file)) {
+        throw new RuntimeException(sprintf('File "%s" doesn\'t exist or is not readable', $file));
+    }
+    $config = Yaml::parse(file_get_contents($file))['config'] ?? [];
+
+    return [
+        $config['exclusions'] ?? [],
+        $config['keepExcludedUsers'] ?? false,
+        $config['extractEmailDomain'] ?? false,
+        $config['fieldsWhitelist'] ? array_flip($config['fieldsWhitelist']) : [],
+        $config['excludeRepositories'] ?? [],
+    ];
+}
+
+(new Application('traces', '2.1.0'))
     ->register('traces')
-    ->setDescription('A simple CLI tool to generate information about GitHub contributors of a repository.')
-    ->addArgument('repository', InputArgument::REQUIRED, 'GitHub repository, for instance PrestaShop/PrestaShop')
+    ->setDescription('A simple CLI tool to generate information about GitHub contributors of an organization.')
+    ->addArgument('organization', InputArgument::REQUIRED, 'GitHub organization, for instance PrestaShop')
```

Comment:
How about keeping the repository option? Let the user choose between repository and organization

Reply:
first step: show the contribution on the org.
then, having a way to show contributions by components would be great. (core, each module, ...)
Also, there is the idea of a contributor profile like API Platform or Drupal, listing on each profile the number of contributions per repository.

Reply:
Done ;) And updated the readme ;)